galasa-extensions-parent/dev.galasa.ecosystem.docker.builder/src/main/java/dev/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java

```diff
@@ -0,0 +1,525 @@
+				"           --detach \\\n" + 
+				"           --mount source=" + cpsMountSource + ",target=" + cpsMountTarget + " \\\n" + 
+				"           --publish " + cpsPort + ":2379 \\\n" + 
+				"           quay.io/coreos/etcd:v3.2.25 \\\n" + 
+				"           etcd --data-dir /var/run/etcd/default.etcd --initial-cluster default=http://" + hostname + ":2380 --listen-client-urls http://0.0.0.0:2379 --listen-peer-urls http://0.0.0.0:2380 --initial-advertise-peer-urls http://127.0.0.1:2380 --advertise-client-urls http://127.0.0.1:2379\n";
+		String rasCont = "docker run --name " + rasContainerName + " \\\n" + 
+				"           --network " + networkName + " \\\n" + 
+				"           --restart " + resPol + " \\\n" + 
+				"           --detach \\\n" + 
+				"           --env COUCHDB_USER=galasa \\\n" +
+				"           --env COUCHDB_PASSWORD=galasa \\\n" +
+				"           --mount source=" + rasMountSource + ",target=" + rasMountTarget + " \\\n" + 
+				"           --publish " + rasPort + ":5984 \\\n" + 
+				"           couchdb:2 \n" ;
+		String resmonCont = "docker run --name " + resourceMonitorContainerName + " \\\n" + 
+				"           --network " + networkName + " \\\n" + 
+				"           --restart " + resPol + " \\\n" + 
+				"           --detach \\\n" + 
+				"           docker.galasa.dev/galasa-boot-embedded-amd64:" + bootVersion + " \\\n" + 
+				"           java -jar boot.jar --obr file:galasa.obr --resourcemanagement --bootstrap http://" + apiMountSource + ":8181/bootstrap \n";
+		String resCont = "docker run --name " + resourceContainerName + " \\\n" + 
+				"           --network " + networkName + " \\\n" + 
+				"           --restart " + resPol + " \\\n" + 
+				"           --detach \\\n" + 
+				"           --publish " + resourcePort + ":80 \\\n" + 
+				"           docker.galasa.dev/galasa-resources-amd64:" + resourceVersion +  "\n";
+		String sbCont = "docker run --name " + sbName + " \\\n" + 
+				"           --network " + networkName + " \\\n" + 
+				"           --restart " + resPol + " \\\n" + 
+				"           --detach \\\n" + 
+				"           --publish " + sbPort + ":2080 \\\n" + 
+				"           --publish 2023:2023 \\\n" + 
+				"           docker.galasa.dev/galasa-boot-embedded-amd64:" + bootVersion + " \\\n" + 
+				"           java -jar simframe.jar";
+		
+		File script = new File("docker-ecosystem-setup.sh");
+		if (script.exists()) {
+			script.delete();
+		}
+		
+		PrintWriter writer = new PrintWriter("docker-ecosystem-setup.sh", "UTF-8");
+		writer.println("#!/bin/bash \n");
+		writer.println(dockerNet);
+		writer.println(volCreate);
+		writer.println(apiCont);
+		writer.println(contCont);
+		writer.println(cpsCont);
+		writer.println(rasCont);
+		writer.println(resmonCont);
+		writer.println(resCont);
+		writer.println(sbCont);
+		writer.close();
+
+		System.out.println("Script generated, updating permissions... \n");
+		
+		try {
+			Runtime.getRuntime().exec("chmod u+x docker-ecosystem-setup.sh");
```

Comment:
You have made an assumption this is running on linux or mac.   will fail if on windows.   delete, not necessary,  they can handle the the permissions or do "sh docker-ecosystem-setup.sh"